src/app/components/ModeModal.ts

```diff
@@ -155,63 +151,150 @@ export class ModeModal extends BasicModal {
+            { className: 'mode-form__color-wrapper' },
+            label,
+            radio,
+          ),
+        );
+        return colorsElements;
+      },
+      [],
+    );
+    const colors = Render.elementFactory(
+      'div',
+      { className: 'mode-form__colors' },
+      ...colorInputs,
+    );
+
+    return colors;
+  }
+
+  private generateAvatarInput(numberOfPlayer: number): HTMLElement {
+    const avatarInputs: HTMLElement[] = Object.values(Avatars).reduce(
+      (avatarsElements: HTMLElement[], value, index) => {
+        const indicator = `${numberOfPlayer}avatarChoice_${
+          index + 1
+        }`;
+        const label = Render.elementFactory(
+          'label',
+          {
+            for: indicator,
+            className: 'mode-form__avatar-label',
+          },
+          Render.elementFactory('img', {
+            className: 'mode-form__avatar',
+            src: value,
+          }),
+        );
+        const radio = Render.elementFactory('input', {
+          type: 'radio',
+          name: `path_${numberOfPlayer}`,
+          className: 'mode-form__avatar-radio',
+          id: indicator,
+          value: value,
+        });
+        avatarsElements.push(
+          Render.elementFactory(
+            'div',
+            { className: 'mode-form__avatar-wrapper' },
+            label,
+            radio,
+          ),
+        );
+        return avatarsElements;
+      },
+      [],
+    );
+    const colors = Render.elementFactory(
+      'div',
+      { className: 'mode-form__avatars' },
+      ...avatarInputs,
+    );
+
+    return colors;
+  }
+
+  private generateAddPlayerFields(
+    numberOfPlayer: number,
+  ): HTMLElement {
+    const fieldsWrapper = Render.elementFactory(
       'div',
-      { className: 'mode-form__row' },
-      label,
-      input,
+      {},
+      this.generateNameInput(numberOfPlayer),
+      this.generateAvatarInput(numberOfPlayer),
+      this.generateColorInput(numberOfPlayer),
     );
-    return playerRow;
+    return fieldsWrapper;
   }
 
   private convertDataFormToPlayersData(
     formData: FormData,
   ): PlayerDTO[] {
     const playersData = [];
     for (const [formKey, formValue] of formData.entries()) {
-      const playerDTO: PlayerDTO = {
-        name: '',
-        path: './static/images/playerAvatars/woman.svg',
-        color: '#064a89',
-      };
       const value = formValue.toString();
-      const [key] = formKey.split('_');
+      const [key, numberOfPlayer] = formKey.split('_');
+      const index = +numberOfPlayer - 1;
       switch (key) {
         case 'name': {
-          playerDTO.name =
+          playersData.push({
+            name: '',
+            path: Avatars.FARMER1,
+            color: Colors.GREEN,
+          } as PlayerDTO);
+          playersData[index].name =
             value.trim().length > 0
               ? value
-              : `Janush ${playersData.length + 1}`;
+              : `Janush ${playersData.length}`;
```

Comment:
Perhaps the default names should match, or be given in placeholder? (for example 'type nickname here' and then Player1, Player2 on the board, or 'type nickname here, ex.: Janush1')

Reply:
Placeholder will be 'type nickname here' according to figma project